core/src/test/java/org/elasticsearch/index/query/TemplateQueryBuilderTest.java

```diff
@@ -18,21 +18,41 @@
  */
 package org.elasticsearch.index.query;
 
+import org.apache.lucene.search.Query;
 import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.common.xcontent.XContentFactory;
 import org.elasticsearch.script.ScriptService.ScriptType;
 import org.elasticsearch.script.Template;
-import org.elasticsearch.test.ESTestCase;
 import org.junit.Test;
 
 import java.io.IOException;
 import java.util.HashMap;
 import java.util.Map;
 
-/**
- * Test building and serialising a template search request.
- * */
-public class TemplateQueryBuilderTest extends ESTestCase {
+import static org.hamcrest.Matchers.is;
+
+public class TemplateQueryBuilderTest extends BaseQueryTestCase<TemplateQueryBuilder> {
+
+    @Override
+    protected boolean supportsBoostAndQueryName() {
+        return false;
+    }
+
+    @Override
+    protected TemplateQueryBuilder doCreateTestQueryBuilder() {
+        return new TemplateQueryBuilder(new Template(RandomQueryBuilder.createQuery(random()).toString()));
+    }
+
+    @Override
+    protected void doAssertLuceneQuery(TemplateQueryBuilder queryBuilder, Query query, QueryShardContext context) throws IOException {
+        assertEquals(query, queryBuilder.toQuery(context));
+    }
+
+    @Test
+    public void testValidate() {
+        TemplateQueryBuilder templateQueryBuilder = new TemplateQueryBuilder(null);
+        assertThat(templateQueryBuilder.validate().validationErrors().size(), is(1));
+    }
 
     @Test
     public void testJSONGeneration() throws IOException {
```

Comment:
If I'm not mistaken to be able to test the same code paths that are being checked down here the random querybuilder generation above would have to be changed to also include generated parameters.